### PR TITLE
drivers/mtd_mapper: fix read_page and write_page backend

### PR DESCRIPTION
**Pull request by jue89**
### Contribution description

This PR fixes the `mtd_mapper` backend for `read_page` and `write_page`.

The implementation uses `mtd_read_page()` and `mtd_write_page_raw()` on the backing device. These methods return `0` on successful [read](https://github.com/RIOT-OS/RIOT/blob/753b435cb50543a68b43354773c4ff0603ab9f58/drivers/include/mtd.h#L305) resp. [write](https://github.com/RIOT-OS/RIOT/blob/753b435cb50543a68b43354773c4ff0603ab9f58/drivers/include/mtd.h#L353). But the backend is expected to return the amount of [read](https://github.com/RIOT-OS/RIOT/blob/753b435cb50543a68b43354773c4ff0603ab9f58/drivers/include/mtd.h#L166) resp. [written](https://github.com/RIOT-OS/RIOT/blob/753b435cb50543a68b43354773c4ff0603ab9f58/drivers/include/mtd.h#L206) bytes.

### Testing procedure

Run `tests/mtd_mapper`. I've added a test triggering this bug. Without this patch it'll be caught in an endless loop.

### CODE REVIEW

**Review by benpicco — APPROVED** [review]
Good catch, thank you!

**Comment by jue89** [comment]
Thanks for your review!

File: tests/mtd_mapper/Makefile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 include ../Makefile.tests_common
 
 USEMODULE += mtd_mapper
+USEMODULE += mtd_write_page
```
Comment (benpicco) [inline tests/mtd_mapper/Makefile:4]: Kconfig needs an update too - please squash directly

Comment (jue89) [reply tests/mtd_mapper/Makefile:4]: Sure ... always forgetting this.

I hope we'll get rid of this redundancy soon ;-)

Comment (jue89) [reply tests/mtd_mapper/Makefile:4]: Sure ... always forgetting this.

I hope we'll get rid of this redundancy soon ;-)

**Pull request merged**